### PR TITLE
test(fs/walk): walk works with fifo

### PR DESCRIPTION
**Pull request by clemens-tolboom**
Waiting for #1792

As reported in #1789 walk fails on in my case `/Users/clemens/.docker/run/docker.sock` which is a socket.

```
file /Users/clemens/.docker/run/docker.sock
/Users/clemens/.docker/run/docker.sock: socket
```

As I don't know how to create a socket I went for `mkfifo`

### CODE REVIEW

**Comment by lucacasonato** [comment]
See #1792

**Comment by clemens-tolboom** [comment]
Blocked by/waiting on #1792

I made this a `mkfifo` version. #1792 is about socket.

**Comment by clemens-tolboom** [comment]
Not sure why `mkfifo` failed only on Ubuntu ... testing locally my MacOS failed (my bad).

File: fs/walk_test.ts [hunk 1]
```diff
@@ -341,3 +341,16 @@ testWalk(
...
 );
+
+testWalk(
+  (d: string) => {
+    return Deno.run({cmd:['mkfifo', `${d}/mkfifo`]});
+  },
+  async function fifoSocket(listener: Deno.Process) {
+    assertReady(1);
+    const files = await walkArray(".", { followSymlinks: true });
+    assertEquals(files, [".", "mkfifo"]);
```
Comment (clemens-tolboom) [inline fs/walk_test.ts:352]: I guess I have to revert back to expecting only ['.'] as #1792 ignores everything but files, dirs and symlinks iirc.

**Comment by clemens-tolboom** [comment]
Not sure I should have closed this one ... the fix done in #1792 just ignores all non file/dir types (ruling out "sockets treeWalk searches"; imho not ok)

**Comment by kt3k** [comment]
@clemens-tolboom I think the issue was that `Deno.run` call wasn't awaited correctly. I fixed that in https://github.com/denoland/deno_std/pull/1793/commits/fa5f8c59b50c65cd4e92e098ef4464931c71ddb9 (by using Deno.spawn, which is newer version of Deno.run). Now I think this PR is ready to land

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by clemens-tolboom** [comment]
TNX

**Pull request merged**